oraclizeAPI_0.5.sol

```diff
@@ -839,115 +956,140 @@ contract usingOraclize {
+                if (decimals) {
+                   if (_b == 0) break;
+                    else _b--;
+                }
+                mint *= 10;
+                mint += uint(uint8(bresult[i])) - 48;
+            } else if (uint(uint8(bresult[i])) == 46) {
+                 decimals = true;
+            } else {
+                revert("Non-numeral character encountered in string!");
+            }
+        }
+        if (_b > 0) {
+            mint *= 10 ** _b;
+        }
+        return mint;
     }
 
-    // parseInt
-    function parseInt(string _a) internal pure returns (uint) {
+    function parseInt(string memory _a) internal pure returns (uint) {
         return parseInt(_a, 0);
     }
 
-    // parseInt(parseFloat*10^_b)
-    function parseInt(string _a, uint _b) internal pure returns (uint) {
+    function parseInt(string memory _a, uint _b) internal pure returns (uint) {
         bytes memory bresult = bytes(_a);
         uint mint = 0;
         bool decimals = false;
-        for (uint i=0; i<bresult.length; i++){
-            if ((bresult[i] >= 48)&&(bresult[i] <= 57)){
-                if (decimals){
-                   if (_b == 0) break;
-                    else _b--;
+        for (uint i = 0; i < bresult.length; i++) {
+            if ((uint(uint8(bresult[i])) >= 48) && (uint(uint8(bresult[i])) <= 57)) {
+                if (decimals) {
+                   if (_b == 0) {
+                       break;
+                   } else {
+                       _b--;
+                   }
                 }
                 mint *= 10;
-                mint += uint(bresult[i]) - 48;
-            } else if (bresult[i] == 46) decimals = true;
+                mint += uint(uint8(bresult[i])) - 48;
+            } else if (uint(uint8(bresult[i])) == 46) {
+                decimals = true;
+            }
+        }
+        if (_b > 0) {
+            mint *= 10 ** _b;
         }
-        if (_b > 0) mint *= 10**_b;
         return mint;
     }
 
-    function uint2str(uint i) internal pure returns (string){
-        if (i == 0) return "0";
-        uint j = i;
+    function uint2str(uint _i) internal pure returns (string memory) {
+        if (_i == 0) {
+            return "0";
+        }
+        uint j = _i;
         uint len;
-        while (j != 0){
+        while (j != 0) {
             len++;
             j /= 10;
         }
         bytes memory bstr = new bytes(len);
         uint k = len - 1;
-        while (i != 0){
-            bstr[k--] = byte(48 + i % 10);
-            i /= 10;
+        while (_i != 0) {
+            bstr[k--] = byte(uint8(48 + _i % 10));
+            _i /= 10;
         }
         return string(bstr);
     }
 
-    using CBOR for Buffer.buffer;
-    function stra2cbor(string[] arr) internal pure returns (bytes) {
+    function stra2cbor(string[] memory _arr) internal pure returns (bytes memory) {
         safeMemoryCleaner();

```

Comment:
Should be droppable, with when this was implemented, there may be changes in upcoming versions that could break from this, if they add dirty memory (memory in the freemem region, where the assumption that the empty memory is empty, is not true, albeit if it's there in the first place, it shouldn't affect anything if it's cleared, if the compiler works what I think is properly)